### PR TITLE
Batch starter updates

### LINKED ISSUE #250: Batch Starter Lambda update post February demo
batch_starter.py updates: 
- Update `--dependency <data>` to send result from query API and may be filter out `file_path` from query result.
- Update to use PreProcessingDependency table instead of .json file.
- Update to not construct `file_path_to_create` and instead change `--file_path` to `input_file_path`(discuss it).
- Update event input and rule to match changes from above.
- How to track version and coordinate with imap_processing repo.

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary

Started out as batch starter lambda update but had to update others to line up with changes from batch stater lambda. Tested in AWS and it works as expected.

Main changes:

- sds_data_manager/lambda_code/SDSCode/batch_starter.py
   - updated batch command input and as a result update event bridge input too
- sds_data_manager/lambda_code/SDSCode/database/models.py
   - updated status table
- sds_data_manager/lambda_code/SDSCode/indexer.py
   - updated indexer handler to handle changes from batch starter lambda
- sds_data_manager/lambda_code/SDSCode/upload_api.py
   - found some bugs and fixed it

### CODE REVIEW

**Review by maxinelasp — APPROVED** [review]
Nice, I think this looks good! Thanks for the quick implementation!

**Comment by tech3371** [comment]
> Nice, I think this looks good! Thanks for the quick implementation!

No problem! I wanted to get this in before so that it unblocks your work and you can enjoy your vacation.

**Comment by tech3371** [comment]
@laspsandoval Before I merge this, can you review this PR when you get chance since you wrote most of these?

**Review by bourque — APPROVED** [review]
Looks good!

File: sds_data_manager/lambda_code/SDSCode/batch_starter.py
Comment (laspsandoval) [inline sds_data_manager/lambda_code/SDSCode/batch_starter.py:307]: Small change: use either 'level' or 'data_level'.

Comment (tech3371) [reply sds_data_manager/lambda_code/SDSCode/batch_starter.py:307]: I was concerned about that too. batch job command(aka imap_cli) uses `level` and all the other places uses`data_level`. @maxinelasp can you update `imap_cli` to use `data_level` as well and I can change it here.

File: sds_data_manager/lambda_code/SDSCode/batch_starter.py
Comment (laspsandoval) [inline sds_data_manager/lambda_code/SDSCode/batch_starter.py:481]: Any way to change this to where it isn't a hard-coded int?

**Review by laspsandoval — APPROVED** [review]
Looks so much better. Thanks for doing this Tenzin. Just a couple of nit-picky comments.

File: sds_data_manager/lambda_code/SDSCode/batch_starter.py
Comment (tech3371) [reply sds_data_manager/lambda_code/SDSCode/batch_starter.py:481]: Not for this because we need to send batch command as list of string. We could think about this better way to do this down the road.

File: sds_data_manager/lambda_code/SDSCode/batch_starter.py
Comment (maxinelasp) [reply sds_data_manager/lambda_code/SDSCode/batch_starter.py:307]: I think I already made that change yesterday? Let me double check

Comment (maxinelasp) [reply sds_data_manager/lambda_code/SDSCode/batch_starter.py:307]: I changed dependency to `data_level` but not the CLI itself. Now, everything should be data_level. Thanks for catching that, it's great to get all these names sorted out now!

Comment (tech3371) [reply sds_data_manager/lambda_code/SDSCode/batch_starter.py:307]: Yes. Thank you both so much!

Comment (tech3371) [reply sds_data_manager/lambda_code/SDSCode/batch_starter.py:307]: I updated places where I can from `level` to `data_level`. I didn't change it in places where we use `level` when we build dependency. We will need to update that in upcoming work. so I left it as it is for now.

Comment (tech3371) [reply sds_data_manager/lambda_code/SDSCode/batch_starter.py:307]: I decided to change all afterwards :) and tested in AWS. now the batch job command looks like this 
```
--instrument swe --data_level l1a --start-date 20230927 --end-date 20230927 --version v03-00 --dependency [{'instrument': 'swe', 'data_level': 'l0', 'version': 'v03-00'}] --use-remote
```

**Pull request merged**
